Classes/Netlogix/JsonApiOrg/Consumer/Guzzle/Middleware/EndpointCacheMiddleware.php

```diff
@@ -0,0 +1,96 @@
+<?php
+declare(strict_types=1);
+
+namespace Netlogix\JsonApiOrg\Consumer\Guzzle\Middleware;
+
+use GuzzleHttp\Promise\FulfilledPromise;
+use GuzzleHttp\Promise\Promise;
+use GuzzleHttp\Psr7\Response;
+use GuzzleHttp\Psr7\Utils;
+use Neos\Cache\Frontend\VariableFrontend;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+class EndpointCacheMiddleware
+{
+
+    protected array $httpMethods = ['GET' => true];
+
+    /**
+     * @var VariableFrontend
+     * @Flow\Inject
+     */
+    protected $cache;
+
+    public function __invoke(callable $handler)
+    {
+        return function (RequestInterface $request, array $options) use (&$handler) {
+            if (!isset($this->httpMethods[strtoupper($request->getMethod())])) {
+                // No caching for this method allowed
+                return $handler($request, $options);
+            }
+
+            if (!self::str_ends_with(rtrim($request->getUri()->getPath(), '/'), '/.well-known/endpoint-discovery')) {
+                return $handler($request, $options);
+            }
+
+            $cacheIdentifier = self::getCacheKey($request);
+
+            if ($this->cache->has($cacheIdentifier)) {
+                [$body, $headers] = $this->cache->get($cacheIdentifier);
+                $response = new Response(200, $headers, $body, '1.1');
+                return new FulfilledPromise(
+                    $response
+                );
+            }
+
+            $promise = $handler($request, $options);
+            assert($promise instanceof Promise);
+
+            return $promise->then(
+                function (ResponseInterface $response) use ($request, $cacheIdentifier) {
+                    if ($response->getStatusCode() !== 200) {
+                        return $response;
+                    }
+
+                    $body = $response->getBody();
+
+                    // If the body is not seekable, we have to replace it by a seekable one
+                    if (!$body->isSeekable()) {
+                        $response = $response->withBody(
+                            Utils::streamFor($body->getContents())
+                        );
+                    }
+
+                    $this->cache->set($cacheIdentifier, [
```

Comment:
I guess at some point we need to flush this cache when remotes send notifications e.g. about deployments. Should we, as a default, add every type name the result body provides as valid cache tag so we can for example `flushByTags(["person-person", "person-address])`?